docs/robot.md

```diff
@@ -0,0 +1,136 @@
+- Addresses
+  - We add the Hostname and (depending on the configuration & availability) the IPv4 and IPv6 addresses of the server in `Node.status.addresses`.
+  - For the IPv6 address we use the first address in the Network -> For the network `2a01:f48:111:4221::` we add the address `2a01:f48:111:4221::1`.
+  - Private IPs in a vSwitch are not supported.
+
+### Node Lifecycle Controller
+
+The Node Lifecycle Controller is responsible for updating the shutdown status of Nodes & deleting the Kubernetes Node object if the corresponding server is removed.
+
+Both are generally supported. The shutdown status can only be detected if the Robot Server supports this.
+
+### Service Controller (Load Balancers)
+
+The service controller watches Services with `type: LoadBalancer` and creates Cloud Load Balancers for them. By default, all Kubernetes Nodes including Robot servers are added as targets to the Load Balancer. Check out the [Load Balancer Documentation](./load_balancers.md) for more details.
+
+### Unsupported
+
+#### Routes & Private Networks
+
+Adding support for Routing Pod CIDRs through the (Cloud) Networks & (Robot) vSwitches is not currently supported. You will need to use your own CNI for this. 
+
+If you are interested in this, we are looking for contributors to help design & implement this.
+
+## Requirements
+
+### Identifying the correct Server
+
+When a new Node joins the cluster, we first need to figure out which Robot (or Cloud) Server matches this node. We primarily try to match this through the Node Name & the Name of the server in Robot. If you use Kubeadm, the Node Name by default is the Hostname of the server.
+
+_This means that by default, your **Hostname** needs to be the same of the **name of the server in Robot**_. If this does not match, we can not properly match the two entities. Once we have made this connection, we save the Robot Server Number to the field `spec.providerId` on the Node, and use this identifier for any further processing.
+
+If you absolutely need to use different names in Robot & Hostname, you can also configure the Provider ID yourself. This can be done on the `kubelet` through the flag [`--provider-id`](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/). You need to follow the format `hrobot://$SERVER_NUMBER` when setting this. If this format is not followed exactly we can not process this node.
+
+## Config Options
+
+### Credentials
+
+You need to add your Robot credentials into the secret `hcloud`:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: hcloud
+  namespace: kube-system
+type: Opaque
+stringData:
+  token: your-hcloud-token
+  robot-user: your-robot-user-name
+  robot-password: your-robot-password
+```
+
+If you only plan to use a single Robot server, you can also use an "Admin login" for this server instead of the account credentials.
```

Comment:
What does "Admin login" refer to ? Do we have link?

Reply:
Its a concept in the Robot product. You can create a unique username+password that can only access a single server. There is no public documentation for this besides one mention [here](https://docs.hetzner.com/robot/general/overview).

Its what we use for the e2e tests, to avoid passing the full account credentials into the pipeline.

Reply:
![screenshot-admin-login](https://github.com/hetznercloud/hcloud-cloud-controller-manager/assets/3393079/8612ea84-b677-487f-95e0-da1c109551c5)

Screenshot from robot.hetzner.com/server

Reply:
```suggestion
If you only plan to use a single Robot server, you can also use an "Admin login" (see the `Admin login` tab on the server administration page) for this server instead of the account credentials.
```